util/importer.py

```diff
@@ -8,7 +8,13 @@
 from django.conf import settings
 
 
-def import_path(path: str, module_name: str = "", force_reload = False):
+# this is needed so that import from course files works correctly.
+# it seems like importing a module requires the whole dotted path to a
+# path in sys.path, otherwise imports within the module's package fail
+sys.path.append(settings.COURSES_PATH)
```

Comment:
Could you give an example of what kind imports this is about?

For example, was this about JSAV exercises that define a custom view type (Python function in the course repo)? MOOC-Grader exercise view would then import the Python module in the course repo.

Is there a reason why `sys.path` is modified in this module specifically? Could this be done in settings.py?

Reply:
Why did the imports work before v1.12 without this sys.path change?

Reply:
It's about any custom view import. E.g. the JSAV exercises. They worked before because it was assumed that COURSES_PATH is courses/ but it isn't anymore. The modification should be possible in settings.py as well but I put it here as it pertains to the importing functionality that this file is about. E.g. when reading the source for the import functions, I think it is useful to have this nearby.

Reply:
OK. Is it guaranteed that `sys.path.append()` is executed only once even when the module `util/importer.py` is imported in many other modules?

Reply:
This is guaranteed, at least normally. It is possible that Django, uWSGI or some other framework does something unexpected. I added an if check just in case.